packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/ScmTypeFactory.cs

```diff
@@ -23,6 +24,18 @@ public class ScmTypeFactory : TypeFactory
 
         public virtual CSharpType TokenCredentialType => throw new NotImplementedException("Token credential is not supported in Scm libraries yet");
 
+        public virtual Type ClientResponseType => typeof(ClientResult);
+
+        public virtual Type ClientResponseOfTType => typeof(ClientResult<>);
+
+        public virtual Type ClientResponseExceptionType => typeof(ClientResultException);
+
+        public virtual Type HttpResponseType => typeof(PipelineResponse);
```

Comment:
could we not having these but use those api abstraction to represent the types? This looks duplicated because those abstractions already have those types defined there.

Reply:
I think we need both, the actual type is needed for return types and static methods callers.